src/femr/labelers/omop_inpatient_admissions.py

```diff
@@ -109,9 +109,17 @@ def get_outcome_times(self, patient: Patient) -> List[datetime.datetime]:
     def get_prediction_times(self, patient: Patient) -> List[datetime.datetime]:
         """Return end of admission as prediction timm."""
         times: List[datetime.datetime] = []
-        for __, discharge_time in get_inpatient_admission_discharge_times(patient, self.ontology):
+        prev_discharge_date: datetime.date = datetime.date(1900, 1, 1)
+
+        for admission_time, discharge_time in get_inpatient_admission_discharge_times(patient, self.ontology):
             prediction_time: datetime.datetime = self.prediction_time_adjustment_func(discharge_time)
+
+            # Ignore patients who are readmitted the same day they were discharged b/c of data leakage
+            if admission_time.date() <= prev_discharge_date:
```

Comment:
This might seem stupid, but which isn't this just 

```
if prediction_time >= discharge_time:
    continue
```

Reply:
For readmission predictions, the prediction time is by default 11:59 PM on discharge date, and so it should always be >= discharge_time. Instead, we want to exclude any re-admissions that occurred on the same date as discharge, since these re-admissions would've occurred before or on prediction time.

Reply:
Oh, I see. Good catch and thanks for finding it. This should also probably be made general across all labelers, just not in this PR.